src/Build/Evaluation/Conditionals/FunctionCallExpressionNode.cs

```diff
@@ -38,20 +38,11 @@ internal override bool BoolEvaluate(ConditionEvaluator.IConditionEvaluationState
                 // Check we only have one argument
                 VerifyArgumentCount(1, state);
 
-                // Expand properties and items, and verify the result is an appropriate scalar
-                string expandedValue = ExpandArgumentForScalarParameter("exists", (GenericExpressionNode)_arguments[0], state);
-
-                if (String.IsNullOrEmpty(expandedValue))
-                {
-                    return false;
-                }
-
                 try
                 {
-                    if (state.EvaluationDirectory != null && !Path.IsPathRooted(expandedValue))
-                    {
-                        expandedValue = Path.GetFullPath(Path.Combine(state.EvaluationDirectory, expandedValue));
-                    }
+                    return !ExpandArgumentAsFileList((GenericExpressionNode) _arguments[0], state)
+                        .DefaultIfEmpty()
+                        .Any(i => i == null || state.LoadedProjectsCache?.TryGet(i) == null && !FileUtilities.FileOrDirectoryExistsNoThrow(i));
```

Comment:
In `Exists("foo.cs;bar.cs")` I'd expect Exists to check that both files exist.

---
In reply to: [113551386](https://github.com/Microsoft/msbuild/pull/2014#discussion_r113551386) [](ancestors = 113551386)